test/test_jit.py

```diff
@@ -12037,6 +12037,37 @@ def fn(x=None):
         self.checkScript(fn, ((3, 4),))
         self.checkScript(fn, ())
 
+    def _test_pickle_checkpoint(self, device):
```

Comment:
This needs more testing. I don't see the common corner case tests for serializing a tensor here. There should be at least:

1. one tensor used twice and a check it comes back as the same tensor
2. one tensor on the GPU, making sure it makes it back to the GPU
3. two tensors that are views of the same tensor, do they end up linked on reload.

Reply:
I added a test for these cases, though there is a caveat that they must be stored with the same call to `torch.save()` for views to work since there's no sharing between different binaries.